api/v1alpha1/nodepool_types.go

```diff
@@ -211,3 +214,33 @@ type Filter struct {
 	// Values includes one or more filter values. Filter values are case-sensitive.
 	Values []string `json:"values"`
 }
+
+// Volume encapsulates the configuration options for the storage device
+type Volume struct {
+	// Device name
+	// +optional
+	DeviceName string `json:"deviceName,omitempty"`
+
+	// Size specifies size (in Gi) of the storage device.
+	// Must be greater than the image snapshot size or 8 (whichever is greater).
+	// +kubebuilder:validation:Minimum=8
+	Size int64 `json:"size"`
+
+	// Type is the type of the volume (e.g. gp2, io1, etc...).
+	// +optional
+	Type string `json:"type,omitempty"`
```

Comment:
Is it possible or desirable to specify an enum here?

Reply:
I think we decided to keep the string during the meeting.

Reply:
Agree. We can revisit later if necessary. Thanks!